src/Feature/Editor/EditorSurface.re

```diff
@@ -213,16 +202,14 @@ let make =
       bufferHighlights
       definition
       bufferSyntaxHighlights
-      shouldRenderWhitespace
-      shouldRenderIndentGuides
       bottomVisibleLine
-      shouldHighlightActiveIndentGuides
       mode
       isActiveSplit
       gutterWidth
       bufferWidthInCharacters={layout.bufferWidthInCharacters}
+      config
     />
-    {showMinimap
+    {Config.Minimap.enabled(config)
```

Comment:
Is there any performance impact to this, versus what we had before? For example, does it have to parse JSON everytime we ask for `Config.Minimap.enabled`? I'd be curious if we are getting any allocations here. If so, perhaps it'd be possible to memoize this.

Reply:
There's definitely some overhead, so I certainly wouldn't recommend putting these into a tight loop. It doesn't need to parse the entire JSON, but does need to look it up by key (which is primarily by a precomputed hash, but also matches the string itself when the hash matches match in case of collision) and decode the value (which for primitive types is just a pattern match).

To avoid this overhead I think we could create a local configuration record that's memoized based on physical equality for the configuration collection. I'll do some testing to see if it's worth the bother though.

Reply:
Results are in!

Before this change:

```
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|  BENCHMARK                                                          |  ITERATIONS |  TIME               |  MINOR GC  |  MAJOR GC  |  MINOR ALLOC  |  PROMOTED   |  MAJOR ALLOC  |
|---------------------------------------------------------------------+-------------+---------------------+------------+------------+---------------+-------------+---------------|
|  EditorSurface - Rendering: Minimal state                           |  10         |  0.251815080643     |  6261      |  0         |  9172453      |  254607     |  2333952      |
|---------------------------------------------------------------------+-------------+---------------------+------------+------------+---------------+-------------+---------------|
|  EditorSurface - Rendering: 1000 Lines state                        |  10         |  0.176056146622     |  6261      |  0         |  8964854      |  103670     |  2181990      |
|---------------------------------------------------------------------+-------------+---------------------+------------+------------+---------------+-------------+---------------|
|  EditorSurface - Rendering: 1000 Lines state - With indent guides   |  10         |  0.175040006638     |  6261      |  0         |  8964924      |  103670     |  2181990      |
|---------------------------------------------------------------------+-------------+---------------------+------------+------------+---------------+-------------+---------------|
|  EditorSurface - Rendering: 100000 Lines state                      |  10         |  0.18078494072      |  6261      |  0         |  8964854      |  103670     |  2181990      |
|---------------------------------------------------------------------+-------------+---------------------+------------+------------+---------------+-------------+---------------|
|  EditorSurface - Rendering: 100000 Lines state - No minimap         |  10         |  0.097088098526     |  1451      |  0         |  2536854      |  62802      |  551452       |
```

After this change:

```
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|  BENCHMARK                                                          |  ITERATIONS |  TIME               |  MINOR GC  |  MAJOR GC  |  MINOR ALLOC  |  PROMOTED   |  MAJOR ALLOC  |
|---------------------------------------------------------------------+-------------+---------------------+------------+------------+---------------+-------------+---------------|
|  EditorSurface - Rendering: Minimal state                           |  10         |  0.248698949814     |  6261      |  0         |  9169698      |  254547     |  2290072      |
|---------------------------------------------------------------------+-------------+---------------------+------------+------------+---------------+-------------+---------------|
|  EditorSurface - Rendering: 1000 Lines state                        |  10         |  0.178162097931     |  6261      |  0         |  8962099      |  103610     |  2138110      |
|---------------------------------------------------------------------+-------------+---------------------+------------+------------+---------------+-------------+---------------|
|  EditorSurface - Rendering: 1000 Lines state - With indent guides   |  10         |  0.176610946655     |  6261      |  0         |  8962698      |  103730     |  2138230      |
|---------------------------------------------------------------------+-------------+---------------------+------------+------------+---------------+-------------+---------------|
|  EditorSurface - Rendering: 100000 Lines state                      |  10         |  0.183979034424     |  6261      |  0         |  8962099      |  103610     |  2138110      |
|---------------------------------------------------------------------+-------------+---------------------+------------+------------+---------------+-------------+---------------|
|  EditorSurface - Rendering: 100000 Lines state - No minimap         |  10         |  0.10581612587      |  1451      |  0         |  2534758      |  62862      |  551512       |
```

So it's actually an improvement, although not a significant one. 2% less major allocation for 1000 lines is the largest difference from the numbers I checked.

Reply:
Interesting. Those are great results! Thanks for checking.

Reply:
After the last refactor:

```
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|  BENCHMARK                                                          |  ITERATIONS |  TIME               |  MINOR GC  |  MAJOR GC  |  MINOR ALLOC  |  PROMOTED   |  MAJOR ALLOC  |
|---------------------------------------------------------------------+-------------+---------------------+------------+------------+---------------+-------------+---------------|
|  EditorSurface - Rendering: Minimal state                           |  10         |  0.256767034531     |  6261      |  0         |  9172609      |  254597     |  2290122      |
|---------------------------------------------------------------------+-------------+---------------------+------------+------------+---------------+-------------+---------------|
|  EditorSurface - Rendering: 1000 Lines state                        |  10         |  0.190254926682     |  6261      |  0         |  8965010      |  103660     |  2138160      |
|---------------------------------------------------------------------+-------------+---------------------+------------+------------+---------------+-------------+---------------|
|  EditorSurface - Rendering: 1000 Lines state - With indent guides   |  10         |  0.191276073456     |  6261      |  0         |  8966318      |  103950     |  2138450      |
|---------------------------------------------------------------------+-------------+---------------------+------------+------------+---------------+-------------+---------------|
|  EditorSurface - Rendering: 100000 Lines state                      |  10         |  0.197165966034     |  6261      |  0         |  8965010      |  103660     |  2138160      |
|---------------------------------------------------------------------+-------------+---------------------+------------+------------+---------------+-------------+---------------|
|  EditorSurface - Rendering: 100000 Lines state - No minimap         |  10         |  0.11461186409      |  1451      |  0         |  2538268      |  63172      |  551822       |
```

Minor allocations have jumped back up. Major allocations still 2% lower. Still nothing significant.